message.go

```diff
@@ -465,3 +505,270 @@ func (m *Message) Media() Media {
+
+	// New list of reaction types that have been set by the user.
+	NewReaction []react.ReactionType `json:"new_reaction"`
+}
+
+func (mu *MessageReaction) Time() time.Time {
+	return time.Unix(mu.DateUnixtime, 0)
+}
+
+// MessageReactionCount represents reaction changes on a message with
+// anonymous reactions.
+type MessageReactionCount struct {
+	// The chat containing the message.
+	Chat *Chat `json:"chat"`
+
+	// Unique message identifier inside the chat.
+	MessageID int `json:"message_id"`
+
+	// Date of the change in Unix time.
+	DateUnixtime int64 `json:"date"`
+
+	// List of reactions that are present on the message.
+	Reactions *react.ReactionCount `json:"reactions"`
+}
+
+// Time returns the moment of change in local time.
+func (mc *MessageReactionCount) Time() time.Time {
+	return time.Unix(mc.DateUnixtime, 0)
+}
+
+// TextQuote contains information about the quoted part of a message that is
+// replied to by the given message.
+type TextQuote struct {
+	// Text of the quoted part of a message that is replied to by the given message.
+	Text string `json:"text"`
+
+	// (Optional) Special entities that appear in the quote.
+	// Currently, only bold, italic, underline, strikethrough, spoiler,
+	// and custom_emoji entities are kept in quotes.
+	Entities []MessageEntity `json:"entities"`
+
+	// Approximate quote position in the original message in UTF-16 code units
+	// as specified by the sender.
+	Position int `json:"position"`
+
+	// (Optional) True, if the quote was chosen manually by the message sender.
+	// Otherwise, the quote was added automatically by the server.
+	Manual bool `json:"is_manual"`
+}
+
+// MessageOrigin a message reference that has been sent originally by a known user.
+type MessageOrigin struct {
+	// Type of the message origin, always “channel”.
+	Type string `json:"type"`
+
+	// Date the message was sent originally in Unix time.
+	DateUnixtime int64 `json:"date"`
+
+	// User that sent the message originally.
+	Sender *User `json:"sender_user,omitempty"`
+
+	// Name of the user that sent the message originally.
+	SenderUsername string `json:"sender_user_name,omitempty"`
+
+	// Chat that sent the message originally.
+	SenderChat *Chat `json:"sender_chat,omitempty"`
+
+	// Channel chat to which the message was originally sent.
+	Chat *Chat `json:"chat,omitempty"`
+
+	// Unique message identifier inside the chat.
+	MessageID int `json:"message_id,omitempty"`
+
+	// (Optional) For messages originally sent by an anonymous chat administrator,
+	// original message author signature.
+	Signature string `json:"author_signature,omitempty"`
+}
+
+// Time returns the moment of message that was sent originally in local time.
+func (mo *MessageOrigin) Time() time.Time {
+	return time.Unix(mo.DateUnixtime, 0)
+}
+
+// ExternalReply contains information about a message that is being replied to,
+// which may come from another chat or forum topic.
+type ExternalReply struct {
```

Comment:
It feels to me more like `External[Reply]Message` despite the original name saying `ExternalReplyInfo`. The structure is basically a subset of the `Message` content, and working with it means you're accessing the message being replied to.

What do you think?